proxy/backendtimeout_test.go

```diff
@@ -69,6 +73,13 @@ func TestFastService(t *testing.T) {
 }
 
 func TestBackendTimeoutInTheMiddleOfServiceResponse(t *testing.T) {
+	var logBuf bytes.Buffer
```

Comment:
Maybe we can create a `testLog` type that would encapsulate buffer, take care of output restoration and have `WaitFor` method. It would be use like:
```go

tl := NewTestLog(t) // uses *testing.T to restore logrus output via T.Cleanup
...
if err = tl.WaitFor(msg, 100*time.Millisecond); err != nil {
...
}
```